services/director-v2/src/simcore_service_director_v2/api/routes/computations_tasks.py

```diff
@@ -0,0 +1,173 @@
+
+
+async def analyze_pipeline(
+    project_id: ProjectID,
+    comp_pipelines_repo: CompPipelinesRepository,
+    comp_tasks_repo: CompTasksRepository,
+) -> PipelineInfo:
+    """
+    Loads and validates data from pipelines and tasks tables and
+    reports it back as PipelineInfo
+    """
+
+    # NOTE: Here it is assumed the project exists in comp_tasks/comp_pipeline
+    # get the project pipeline
+    pipeline_at_db: CompPipelineAtDB = await comp_pipelines_repo.get_pipeline(
+        project_id
+    )
+    pipeline_dag: nx.DiGraph = pipeline_at_db.get_graph()
+
+    # get the project task states
+    all_tasks: list[CompTaskAtDB] = await comp_tasks_repo.get_all_tasks(project_id)
+
+    # filter the tasks by the effective pipeline
+    filtered_tasks = [
+        t for t in all_tasks if f"{t.node_id}" in set(pipeline_dag.nodes())
+    ]
+
+    # check that we have the expected tasks
+    if len(filtered_tasks) != len(pipeline_dag):
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail="The tasks referenced by the pipeline are missing",
+        )
+
+    return PipelineInfo(pipeline_dag, all_tasks, filtered_tasks)
+
+
+async def _get_task_log_file(
+    user_id: UserID, project_id: ProjectID, node_id: NodeID
+) -> Optional[AnyUrl]:
+    try:
+
+        log_file_url = await get_service_log_file_download_link(
+            user_id, project_id, node_id, file_link_type=FileLinkType.PRESIGNED
+        )
+
+    except NodeportsException as err:
+        # Unexpected error: Cannot determine the cause of failure
+        # to get donwload link and cannot handle it automatically.
+        # Will treat it as "not available" and log a warning
+        log_file_url = None
+        log.warning(
+            "Failed to get log-file of %s: %s.",
+            f"{user_id=}/{project_id=}/{node_id=}",
+            err,
+        )
+    return log_file_url
+
+
+# ROUTES HANDLERS --------------------------------------------------------------
+
+
+@router.get(
+    "/{project_id}/tasks/-/logfile",
+    summary="Gets computation task logs file after is done",
+    response_model=list[TaskLogFileGet],
+)
+async def get_all_tasks_log_files(
+    user_id: UserID,
+    project_id: ProjectID,
+    comp_pipelines_repo: CompPipelinesRepository = Depends(
+        get_repository(CompPipelinesRepository)
+    ),
+    comp_tasks_repo: CompTasksRepository = Depends(get_repository(CompTasksRepository)),
+) -> list[TaskLogFileGet]:
+    """Returns download links to log-files of each task in a computation.
+    Each log is only available when the corresponding task is done
+    """
+    # gets computation task ids
+    info = await analyze_pipeline(project_id, comp_pipelines_repo, comp_tasks_repo)
+
+    return [
+        TaskLogFileGet(
+            task_id=node_id,
+            download_link=await _get_task_log_file(user_id, project_id, node_id),
+        )
+        for node_id in (t.node_id for t in info.filtered_tasks)
+    ]
```

Comment:
maybe you can use logged_gather here and then return?

Reply:
I would not worry about the parallel requests to storage. In theory storage is capable of handling this. If we see some slowdowns here, it can be a pointer to the fact that storage is blocking when requesting these. I do not expect issues tough.

Reply:
Indeed, as you say: "in theory", therefore my question.